src/components/add-site.tsx

```diff
@@ -117,16 +118,20 @@ export default function AddSite( { className }: AddSiteProps ) {
 						doesPathContainWordPress={ doesPathContainWordPress }
 					>
 						<div className="flex flex-row justify-end gap-x-5 mt-6">
-							<Button onClick={ closeModal } disabled={ isAddingSite } variant="tertiary">
+							<Button
+								onClick={ closeModal }
+								disabled={ selectedSite?.isAddingSite }
+								variant="tertiary"
+							>
 								{ __( 'Cancel' ) }
 							</Button>
 							<Button
 								type="submit"
 								variant="primary"
-								isBusy={ isAddingSite }
-								disabled={ isAddingSite || !! error || ! siteName?.trim() }
+								isBusy={ selectedSite?.isAddingSite }
+								disabled={ selectedSite?.isAddingSite || !! error || ! siteName?.trim() }
 							>
-								{ isAddingSite ? __( 'Adding site…' ) : __( 'Add site' ) }
+								{ __( 'Add site' ) }
```

Comment:
I've noticed that in Onboarding line 125, we are checking for selectedSite?.isAddingSite to display a different string.
I would apply the same decision as above.

```
{ selectedSite?.isAddingSite ? __( 'Adding site…' ) : __( 'Add site' ) }
```